src/main/java/core/basesyntax/BlockingQueue.java

```diff
@@ -12,16 +12,31 @@ public BlockingQueue(int capacity) {
     }
 
     public synchronized void put(T element) throws InterruptedException {
-        // write your code here
+        while (queue.size() == capacity) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Something went wrong", e);
+            }
+        }
+        queue.add(element);
+        notify();
     }
 
     public synchronized T take() throws InterruptedException {
-        // write your code here
-        return null;
+        while (isEmpty()) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Something went wrong", e);
+            }
+        }
+        queue.poll();
```

Comment:
poll() -> retrieves and removes the head of this queue, so when we call queue.peek() we will get another item